backends/occa/ceed-occa-vector.c

```diff
@@ -31,187 +24,154 @@ static inline size_t bytes(const CeedVector vec) {
 
 // *****************************************************************************
 // * OCCA SYNC functions
-// * Ptr == void*, Mem == device
-// * occaCopyPtrToMem(occaMemory dest, const void *src,
-// * occaCopyMemToPtr(void *dest, occaMemory src,
 // *****************************************************************************
 static inline void occaSyncH2D(const CeedVector vec) {
-  const CeedVector_Occa *impl = vec->data;
-  assert(impl);
-  assert(impl->device);
-  occaCopyPtrToMem(*impl->device, impl->array, bytes(vec), NO_OFFSET, NO_PROPS);
-}
-static inline void occaSyncD2H(const CeedVector vec) {
-  const CeedVector_Occa *impl = vec->data;
-  assert(impl);
-  assert(impl->array);
-  assert(impl->device);
-  occaCopyMemToPtr(impl->array, *impl->device, bytes(vec), NO_OFFSET, NO_PROPS);
+  const CeedVector_Occa *occa = vec->data;
+  assert(occa);
+  assert(occa->h_array);
+  assert(occa->d_array);
+  occaCopyPtrToMem(*occa->d_array, occa->h_array, bytes(vec), NO_OFFSET,
+                   NO_PROPS);
 }
-
-// *****************************************************************************
-// * OCCA COPY functions
 // *****************************************************************************
-static inline void occaCopyH2D(const CeedVector vec, void *from) {
-  const CeedVector_Occa *impl = vec->data;
-  assert(from);
-  assert(impl);
-  assert(impl->device);
-  occaCopyPtrToMem(*impl->device, from, bytes(vec), NO_OFFSET, NO_PROPS);
+static inline void occaSyncD2H(const CeedVector vec) {
+  const CeedVector_Occa *occa = vec->data;
+  assert(occa);
+  assert(occa->h_array);
+  assert(occa->d_array);
+  occaCopyMemToPtr(occa->h_array,*occa->d_array, bytes(vec), NO_OFFSET, NO_PROPS);
 }
-//static inline void occaCopyD2H(const CeedVector vec, void *to) {
-// const CeedVector_Occa *impl = vec->data;
-//  assert(to);
-//  assert(impl);
-//  occaCopyMemToPtr(to, *impl->device, bytes(vec), NO_OFFSET, NO_PROPS);
-//}
 
 // *****************************************************************************
-// * Set
+// * Set the array used by a vector,
+// * freeing any previously allocated array if applicable
 // *****************************************************************************
-static int CeedVectorSetArray_Occa(const CeedVector vec,
+static int CeedVectorSetArray_Occa(const CeedVector x,
                                    const CeedMemType mtype,
                                    const CeedCopyMode cmode,
                                    CeedScalar *array) {
-  CeedVector_Occa *impl = vec->data;
+  CeedVector_Occa *data = x->data;
```

Comment:
I'm fine with not calling it `impl`, but why is it called `occa` in some files and `data` in others? The inconsistency just seems whimsical and confusing.